assets/src/components/mapPage.tsx

```diff
@@ -356,9 +369,20 @@ const MapPage = (): ReactElement<HTMLDivElement> => {
                   setFollowerShouldSetZoomLevel(false)
                 })
               }
+              onStartDetour={onStartDetour}
             />
           </MapSafeAreaContext.Provider>
         </div>
+        {originalRouteForDetour && (
+          <DetourModal
+            key={detourModalKey}
```

Comment:
**question (non-blocking):** can you help me understand what necessitates the `detourModalKey`? If `originalRouteForDetour` gets reset to `null` in the `onClose` callback I would think you could avoid it.

Reply:
It fixes this bug 👉 https://github.com/mbta/skate/pull/2377#discussion_r1484798109

TL;DR: Leaflet holds onto state even when a React mindset says that it shouldn't. When the `detourModalKey` changes, that convinces React that it's really a different modal, not the same one reappearing, which then lets React convince Leaflet of the same thing.

Reply:
Got it, thanks.